File: spring-boot/src/main/java/springboot/repository/FollowRepository.java
package springboot.repository;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import springboot.domain.entity.FollowEntity;
import springboot.domain.entity.UserEntity;

public interface FollowRepository extends JpaRepository<FollowEntity, Long> {

    @Query("SELECT f FROM FollowEntity AS f WHERE f.followeeId = :followeeId AND f.followerId = :followerId")
    Optional<FollowEntity> findExistFollow(@Param("followeeId") Long followeeId, @Param("followerId") Long followerId);

    @Query("SELECT f FROM FollowEntity AS f WHERE f.followerEntity = :followerEntity AND f.followeeEntity = :followeeEntity")
    Optional<FollowEntity> findExistFollowByEntity(@Param("followerEntity") UserEntity followerEntity, @Param("followeeEntity") UserEntity followeeEntity);
}

File: spring-boot-kt/src/main/kotlin/real/world/springbootkt/domain/follow/FollowRepository.kt
package real.world.springbootkt.domain.follow

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface FollowRepository : JpaRepository<Follow, Long> {
    fun findByFollowerIdAndFolloweeId(followerId: Long, followeeId: Long): Follow?
    fun findByFollowerIdInAndFolloweeId(followerIdIn: List<Long>, followeeId: Long): List<Follow>
}